File: 백준/Silver/11399. ATM/ATM.py
import sys
people = int(sys.stdin.readline())
time = list(map(int, sys.stdin.readline().split()))
tot = []
time.sort()
sum_ = 0
for i in range(len(time)):
    sum_ += time[i]
    tot.append(sum_)

print(sum(tot))

File: 백준/Silver/11399. ATM/ATM.java
import java.io.*;
import java.util.*;

// 그리디
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        int N = Integer.parseInt(br.readLine());
        int[] p = new int[N];
        StringTokenizer st = new StringTokenizer(br.readLine());
        
        for (int n = 0; n < N; n++) {
            p[n] = Integer.parseInt(st.nextToken());           
        }
        
        Arrays.sort(p);

        int result = 0;
        int sum = 0;
        for (int n = 0; n < N; n++) {
            sum += p[n];
            result += sum;            
        }
        
        System.out.println(result);
    }
}
